awsbestpractices/parsing-error-message/ParsingErrorMessage.java

```diff
@@ -0,0 +1,67 @@
+package example;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.S3Object;
+
+public class ParsingErrorMessage {
+
+    private AmazonS3Client s3Client;
+
+    public String normalLogging() {
+        try {
+            s3Client.getObject("foo", "bar");
+        } catch (AmazonServiceException ase) {
+            // Must not be flagged
+            logger.info("Caught error: " + ase.getMessage());
+        }
+    }
+
+    public String branching() {
+        try {
+            s3Client.getObject("foo", "bar");
+        } catch (AmazonServiceException ase) {
+            // Must be flagged
+            if (ase.getMessage().contains("foo")) {
+                logger.info("one thing");
+            } else {
+                logger.info("another thinking");
+            }
+        }
+    }
+
+    public String otherException() {
+        try {
+            s3Client.getObject("foo", "bar");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is manually checking for the existence of an object. You could use the SDK to automatically check for this instead. Use the [doesObjectExist](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/s3/AmazonS3Client.html#doesObjectExist-java.lang.String-java.lang.String-) SDK operation.